src/main/java/com/github/theholywaffle/teamspeak3/api/wrapper/Client.java

```diff
@@ -28,174 +28,201 @@
-
-	public String getVersion() {
-		return get(ClientProperty.CLIENT_VERSION);
-	}
-
-	public boolean isAway() {
-		return getBoolean(ClientProperty.CLIENT_AWAY);
-	}
-
-	public boolean isChannelCommander() {
-		return getBoolean(ClientProperty.CLIENT_IS_CHANNEL_COMMANDER);
-	}
-
-	/**
-	 * Utility method that does a linear search on the array of server group IDs returned
-	 * by {@link #getServerGroups()} and returns {@code true} if that array contains
-	 * the given server group ID.
-	 *
-	 * @param serverGroupId
-	 * 		the ID of the server group to search for
-	 *
-	 * @return whether this client is a member of the given server group
-	 */
-	public boolean isInServerGroup(int serverGroupId) {
-		int[] serverGroupIds = getServerGroups();
-		for (int s : serverGroupIds) {
-			if (s == serverGroupId) return true;
-		}
-		return false;
-	}
-
-	/**
-	 * Utility method that does a linear search on the array of server group IDs returned
-	 * by {@link #getServerGroups()} and returns {@code true} if that array contains
-	 * the ID of the given server group.
-	 *
-	 * @param serverGroup
-	 * 		the server group to search for
-	 *
-	 * @return whether this client is a member of the given server group
-	 */
-	public boolean isInServerGroup(ServerGroup serverGroup) {
-		return isInServerGroup(serverGroup.getId());
-	}
-
-	public boolean isInputHardware() {
-		return getBoolean(ClientProperty.CLIENT_INPUT_HARDWARE);
-	}
-
-	public boolean isInputMuted() {
-		return getBoolean(ClientProperty.CLIENT_INPUT_MUTED);
-	}
-
-	public boolean isOutputHardware() {
-		return getBoolean(ClientProperty.CLIENT_OUTPUT_HARDWARE);
-	}
-
-	public boolean isOutputMuted() {
-		return getBoolean(ClientProperty.CLIENT_OUTPUT_MUTED);
-	}
-
-	public boolean isPrioritySpeaker() {
-		return getBoolean(ClientProperty.CLIENT_IS_PRIORITY_SPEAKER);
-	}
-
-	public boolean isRecording() {
-		return getBoolean(ClientProperty.CLIENT_IS_RECORDING);
-	}
-
-	public boolean isRegularClient() {
-		return getType() == 0;
-	}
-
-	public boolean isServerQueryClient() {
-		return getType() == 1;
-	}
-
-	public boolean isTalking() {
-		return getBoolean("client_flag_talking");
-	}
+    public Client(Map<String, String> map) {
+        super(map);
+    }
+
+    public boolean canTalk() {
+        return getBoolean(ClientProperty.CLIENT_IS_TALKER);
+    }
+
+    public String getAwayMessage() {
+        return get(ClientProperty.CLIENT_AWAY_MESSAGE);
+    }
+
+    public int getChannelGroupId() {
+        return getInt(ClientProperty.CLIENT_CHANNEL_GROUP_ID);
+    }
+
+    public int getChannelId() {
+        return getInt(ClientProperty.CID);
+    }
+
+    /**
+     * Utility method. This method will return a client uri.
+     * A client uri can be used to reference a client.
+     * Example: [URL=client://5/thisisanuid=~Jesus]Jesus[/URL]
+     *
+     * @return clientUri
+     */
+    public String getClientUri() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("[URL=client://");
```

Comment:
`append` can be chained, e.g. `sb.append("name=").append(clientName).append(";");` It might be possible to group a few of these statements together so this method doesn't get so bulky.